service/controller/v2/resource/drainer/create.go

```diff
@@ -136,6 +136,22 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		}
 
 		for _, p := range podList.Items {
+			if key.IsCriticalPod(p.Name) {
```

Comment:
Really nice approach to only collect the relevant pods. Keeps the code below so much cleaner.